private/buf/bufformat/formatter.go

```diff
@@ -75,7 +83,9 @@ func (f *formatter) P(elements ...string) {
 
 // Space adds a space to the generated output.
 func (f *formatter) Space() {
-	f.WriteString(" ")
+	if !strings.ContainsRune("\x00 \t\n", f.lastWritten) {
```

Comment:
This prevents doubled whitespace from accidentally popping up, now that there are more calls to `f.Space()`. In particular, we explicitly put a space between some tokens, but then we might also try to add a space when printing inline comments, and two spaces looks bad.